### ISSUE TITLE
Don't create onDestroyView in fragment if it inherits one that does the unbind

### ISSUE THREAD

**Issue by bryant1410** [avast/android-butterknife-zelezny#74]
For example, I have a base class that already does this:

```java
@Override
public void onDestroyView() {
    super.onDestroyView();
    ButterKnife.unbind(this);
}
```

But when using this plugin on the subclass, it is overriden with the same code.

**Comment by NikolaDespotoski** [comment]
<pre><code>

            PsiMethod[] onDestroyViewsMethods = mClass.findMethodsByName("onDestroyView", true);
            if (onDestroyViewsMethods.length == 0) {
                StringBuilder method = new StringBuilder();
                method.append("@Override public void onDestroyView() {\n");
                method.append("super.onDestroyView();\n");
                method.append(butterKnife.getCanonicalUnbindStatement());
                method.append("(this);\n");
                method.append("}");

                mClass.add(mFactory.createMethodFromText(method.toString(), mClass));
            } else {
                PsiMethod onDestroyView = onDestroyViewsMethods[onDestroyViewsMethods.length - 1];
                if (!containsButterKnifeInjectLine(onDestroyView, butterKnife.getSimpleUnbindStatement())) {
                    onDestroyView.getBody().addBefore(
                            mFactory.createStatementFromText(butterKnife.getCanonicalUnbindStatement() + "(this);",
                                    onDestroyView.getContainingClass()), onDestroyView.getBody().getLastBodyElement());
                }
            }

</pre><code>


Searching for the onDestroyView using .findMethodsByName() with last param true, in base classes will do the trick. If method was found more than once take the last reference and attempt to inject the unbind code in the class of the last method. If I have time, I'll test this and commit it.

**Comment by destil** [comment]
Hi, that's a valid issue. Would you like to submit a pull request?

**Comment by NikolaDespotoski** [comment]
@destil  Yes, I'd would submit. I should test it once more and try to more neater solution than the one in the comment.

**Comment by siempredelao** [comment]
Any progress on this? :worried:
